src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2841,6 +2841,50 @@ exprt c_typecheck_baset::do_special_functions(
     overflow.type() = bool_typet{};
     return overflow;
   }
+  else if(identifier == CPROVER_PREFIX "enum_is_in_range")
+  {
+    // Check correct number of arguments
+    if(expr.arguments().size() != 1)
+    {
+      error().source_location = expr.source_location();
+      error() << "__enum_is_in_range expects one argument" << eom;
+      std::ostringstream error_message;
+      error_message << expr.source_location().as_string() << ": "
+                    << "__enum_is_in_range expects one argument, "
+                    << expr.arguments().size() << " were provided";
+      throw invalid_source_file_exceptiont{error_message.str()};
+    }
+    // Replace expr with giant boolean that checks all the values
+    // of the enumeration as disjunction
+    // This code is similar to goto_checkt::enum_range_check
+    auto arg1 = expr.arguments()[0];
+    typecheck_expr(arg1);
+    if(
+      auto c_enum_tag_type =
+        type_try_dynamic_cast<c_enum_tag_typet>(arg1.type()))
+    {
+      symbolt enum_type = this->lookup(c_enum_tag_type->get_identifier());
+      const c_enum_typet &c_enum_type = to_c_enum_type(enum_type.type);
```

Comment:
Please use `const c_enum_typet &c_enum_type = follow_tag(*c_enum_tag_type);` instead of explicitly going via the symbol.

Reply:
Fixed.